pkg/prometheus/store.go

```diff
@@ -102,3 +102,15 @@ func AddAPIServerConfigToStore(ctx context.Context, store *assets.Store, namespa
 	}
 	return nil
 }
+
+func AddScrapeClassConfigToStore(ctx context.Context, store *assets.Store, namespace string, scrapeClasses []*monv1.ScrapeClass) error {
+	for i, scrapeClass := range scrapeClasses {
+		if err := store.AddTLSConfig(ctx, namespace, scrapeClass.TLSConfig); err != nil {
+			return fmt.Errorf("scrape class %d: %w", i, err)
```

Comment:
the name should be unique
```suggestion
			return fmt.Errorf("scrape class %q: %w", scrapeClass.Name, err)
```